libs/MVS/SceneTexture.cpp

```diff
@@ -509,8 +530,35 @@ bool MeshTexture::ListCameraFaces(FaceDataViewArr& facesDatas, float fOutlierThr
 		faceMap.create(imageData.height, imageData.width);
 		depthMap.create(imageData.height, imageData.width);
 		RasterMesh rasterer(vertices, imageData.camera, depthMap, faceMap);
+		//Creating mask for the image border
+		rasterer.mask = cv::Mat::zeros(imageData.height, imageData.width, CV_8U);
+		cv::Mat imageCopy = imageData.image.clone();
+		cv::cvtColor(imageCopy, imageCopy, cv::COLOR_BGR2GRAY);
+		for (int i = 0; i < imageCopy.rows; i++) {
+			rasterer.mask.at<uint8_t>(i, 0) = imageCopy.at<uint8_t>(i, 0) == 0;
+			rasterer.mask.at<uint8_t>(i, imageCopy.cols - 1) = imageCopy.at<uint8_t>(i, imageCopy.cols - 1) == 0;
+		}
+		for (int j = 0; j < imageCopy.cols; j++) {
+			rasterer.mask.at<uint8_t>(0, j) = imageCopy.at<uint8_t>(0, j) == 0;
+			rasterer.mask.at<uint8_t>(imageCopy.rows - 1, j) = imageCopy.at<uint8_t>(imageCopy.rows - 1, j) == 0;
+		}
+		for (int i = 1; i < imageCopy.rows - 1; i++) {
+			for (int j = 1; j < imageCopy.cols - 1; j++) {
+				if (imageCopy.at<uint8_t>(i, j) == 0 &&
+					(rasterer.mask.at<uint8_t>(i + 1, j) != 0 || rasterer.mask.at<uint8_t>(i - 1, j) != 0 || rasterer.mask.at<uint8_t>(i, j + 1) != 0 || rasterer.mask.at<uint8_t>(i, j - 1) != 0)) {
+					rasterer.mask.at<uint8_t>(i, j) = 255;
+				}
+				if (imageCopy.at<uint8_t>(imageCopy.rows - 1 - i, imageCopy.cols - 1 - j) == 0 &&
+					(rasterer.mask.at<uint8_t>(imageCopy.rows - 1 - i + 1, imageCopy.cols - 1 - j) != 0 || rasterer.mask.at<uint8_t>(imageCopy.rows - 1 - i - 1, imageCopy.cols - 1 - j) != 0 || rasterer.mask.at<uint8_t>(imageCopy.rows - 1 - i, imageCopy.cols - 1 - j + 1) != 0 || rasterer.mask.at<uint8_t>(imageCopy.rows - 1 - i, imageCopy.cols - 1 - j - 1) != 0)) {
+					rasterer.mask.at<uint8_t>(imageCopy.rows - 1 - i, imageCopy.cols - 1 - j) = 255;
+				}
+			}
+		}
```

Comment:
it is probably a good idea to do a dilate of the mask (see cv::dilate()) to remove pixels at the border of the empty image, due to JPG like artifacts